### PR TITLE
Updated Pipedrive docs

### PR DESCRIPTION
**Pull request by dat-a-man**
please review @AstrakhantsevaAA

### CODE REVIEW

File: docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md [hunk 1]
```diff
@@ -12,74 +12,274 @@ Need help deploying these sources, or figuring out how to run them in your data
...
+| product       | Goods or services that a company sells, which can be associated with deals                 |
+| deal          | Potential sale or transaction that you can track through various stages                    |
+| pipeline      | Visual representation of your sales process, displaying the stages and deals in each stage |
+| stage         | Specific step in a sales process where a deal resides based on its progress                |
+| user          | Individual with a unique login credential who can access and use the platform              |
 
-Pipedrive provides a unique domain name that is generally `[company].pipedrive.com`. For example, if your company name is `dltHub`, then the subdomain name is `dlthub.pipedrive.com`.
 
-## Initialize the pipeline from Pipedrive verified source
+## Grab credentials
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md]: done

File: docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md [hunk 2]
```diff
@@ -12,74 +12,274 @@ Need help deploying these sources, or figuring out how to run them in your data
...
+1. In the `.dlt` folder, there's a file called `secrets.toml`. It's where you store sensitive
+   information securely, like access tokens. Keep this file safe. Here's its format for service
+   account authentication:
+
+   ```toml
+   [sources.pipedrive.credentials]
+   # Note: Do not share this file and do not push it to GitHub!
+   pipedrive_api_key = "PIPEDRIVE_API_TOKEN" # please set me up :)
+   ```
+1. Replace `PIPEDRIVE_API_TOKEN` with the API token you [copied above](#grab-api-token)
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md]: done

File: docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md [hunk 3]
```diff
@@ -12,74 +12,274 @@ Need help deploying these sources, or figuring out how to run them in your data
...
+This code generates resources for each entity in [RECENTS_ENTITIES](https://github.com/dlt-hub/verified-sources/blob/master/sources/pipedrive/settings.py), stores them in endpoints_resources,
+and then loads data from each endpoint to the destination.
 
-1. In the `.dlt` folder, you will find `secrets.toml`, which looks like this:
-```toml
-# Put your secret values and credentials here
-# Note: Do not share this file and do not push it to GitHub!
-pipedrive_api_key = "PIPEDRIVE_API_TOKEN" # please set me up :)
+```python
+ endpoints_resources = {}
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md]: done

File: docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md [hunk 4]
```diff
@@ -12,74 +12,274 @@ Need help deploying these sources, or figuring out how to run them in your data
...
+`primary_key`: Designates "id" as the resource's primary key.
 
-1. Install requirements for the pipeline by running the following command:
-```bash
-pip install -r requirements.txt
+`write_disposition`: New data merges with existing data in the destination.
+
+`entity`: Argument for get_recent_items_incremental.
+
+`**resource_kwargs`: Additional arguments for get_recent_items_incremental.
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md]: done

File: docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md [hunk 5]
```diff
@@ -12,74 +12,274 @@ Need help deploying these sources, or figuring out how to run them in your data
...
+        primary_key="id"
+    )(_get_deals_participants)(pipedrive_api_key)
 ```
+`name`: Names the transformer as "deals_participants".
+
+`write_disposition`: Sets the transformer to merge new data with existing data in the destination.
+
+`primary_key`: Designates "id" as the transformer's primary key.
+
+`(_get_deals_participants)(pipedrive_api_key)`: Invokes the _get_deals_participants function, passing pipedrive_api_key, and yields its result to the transformer.
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md]: done

File: docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md [hunk 6]
```diff
@@ -12,74 +12,274 @@ Need help deploying these sources, or figuring out how to run them in your data
...
+    def _get_pages_for_rename(
+        entity: str, fields_entity: str, pipedrive_api_key: str
+    ) -> Dict[str, Any]:
 ```
 
-3. Use `dlt pipeline pipedrive_pipeline show` to make sure that everything loaded as expected.
\ No newline at end of file
+`pipedrive_api_key`: Authentication token for Pipedrive, configured in ".dlt/secrets.toml".
+
+`_get_pages_for_rename`: Retrieves all pages of data for a given entity and updates the existing fields mapping with new data.
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md]: done

File: docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md [hunk 7]
```diff
@@ -12,74 +12,274 @@ Need help deploying these sources, or figuring out how to run them in your data
...
+    ```
+
+1. To load data from a start date:
+
+    ```python
+    # Configure a source for 'activities' starting from the specified date.
+    # The 'custom_fields_mapping' is incorporated to convert custom field hashes into their respective names.
+    activities_source = pipedrive_source(
+        since_timestamp="2023-03-01 00:00:00Z"
+        ).with_resources("activities", "custom_fields_mapping")
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md]: done

File: docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md [hunk 8]
```diff
@@ -12,74 +12,274 @@ Need help deploying these sources, or figuring out how to run them in your data
...
+1. After running this command, a new directory will be created with the necessary files and
+   configuration settings to get started.
+
+For more information, read the
+[Walkthrough: Add a verified source.](../../walkthroughs/add-a-verified-source)
+
+### Add credentials
+
+1. In the `.dlt` folder, there's a file called `secrets.toml`. It's where you store sensitive
+   information securely, like access tokens. Keep this file safe. Here's its format for service
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md]: done

File: docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md [hunk 9]
```diff
@@ -12,74 +12,274 @@ Need help deploying these sources, or figuring out how to run them in your data
...
+
+### Default endpoints
+
+You can write your own pipelines to load data to a destination using this verified source. However,
+it is important to note the complete list of the default endpoints given in
+[pipedrive/settings.py.](https://github.com/dlt-hub/verified-sources/blob/master/sources/pipedrive/settings.py)
+
+In the file "settings.py" you'll find "ENTITY_MAPPINGS" and "RECENTS_ENTITIES".
+
+  - `ENTITY_MAPPING`: is a list detailing entities in Pipedrive. Each tuple contains the entity's name, its associated custom fields, and any additional configuration.Provides dynamic access to Pipedrive entities and custom fields.
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/pipedrive.md]: done

**Review by AstrakhantsevaAA — APPROVED** [review]

**Pull request merged**